### PR TITLE
Feature/cards

### PR DESCRIPTION
**Pull request by RomiKMoveo**
created a generic card component and adjust it to 3 different styles.

created a generic card component.
created 3 components that use the generic card component.

none.

I have:
- [ ] deleted all irrelevant comments, logs and debuggers.
- [ ] assure that I don't print full error stacks, full body response or headers from the server in the console.
- [ ] deleted all sensitive information from the code because it's being imported from environment variables.
- [ ] encrypted keys/passwords that are sent through the internet.
- [ ] sanitized all user inputs to prevent attacks.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: epicure-frontend/src/shared/components/Button/Button.scss [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
-    display: block;
-    width: 21px;
-    height: 24px;
+
+  button {
+    border: none;
+    box-sizing: border-box;
+    cursor: pointer;
+    @include button();
+    height: 1000px;
```
Comment (MoranOshiaMoveo) [inline epicure-frontend/src/shared/components/Button/Button.scss:18]: why needed ?

File: epicure-frontend/src/pages/home/components/Restaurant/Restaurant.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  return (
+    <>
+      <div className="restaurant-container">
+        <div className='restaurant-title'>popular restaurant in epicure:</div>
+        <div className="card-container">
+        <Card image={restaurant.image} title={restaurant.title} type={restaurant.type} chefName={restaurant.chefName} stars={restaurant.stars} ></Card>
+        </div>
+        <Button text="All Restaurants" height="35px" icon={allCardsIcon} color="primary-color-black" backGroundColor=" var(--transparent-color)" onClick={handleClick} children={undefined} ></Button>
```
Comment (MoranOshiaMoveo) [inline epicure-frontend/src/pages/home/components/Restaurant/Restaurant.tsx:29]: it seems that u dont need the children

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: epicure-frontend/src/pages/home/components/Restaurant/Restaurant.tsx (hunk 2 above)
Comment (RomiKMoveo) [reply epicure-frontend/src/pages/home/components/Restaurant/Restaurant.tsx:29]: done

File: epicure-frontend/src/shared/components/Button/Button.scss (hunk 1 above)
Comment (RomiKMoveo) [reply epicure-frontend/src/shared/components/Button/Button.scss:18]: done

**Pull request merged**